core/metrics-core-service/src/main/java/org/hawkular/metrics/core/service/TempTablesCleaner.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright 2014-2018 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.hawkular.metrics.core.service;
+
+import java.util.concurrent.TimeUnit;
+
+import org.hawkular.rx.cassandra.driver.RxSession;
+import org.jboss.logging.Logger;
+import org.joda.time.Days;
+
+import com.datastax.driver.core.PreparedStatement;
+
+import rx.Observable;
+import rx.schedulers.Schedulers;
+
+/**
+ * @author jsanda
+ */
+public class TempTablesCleaner {
+
+    private static Logger logger = Logger.getLogger(TempTablesCleaner.class);
+
+    private RxSession session;
+
+    private PreparedStatement findTables;
+
+    private long ttl;
+
+    private DataAccessImpl dataAccess;
+
+    private volatile boolean finished;
+
+    private static final String DROP_TABLE_CQL = "DROP TABLE IF EXISTS %s";
+
+    public TempTablesCleaner(RxSession session, DataAccessImpl dataAccess, String keyspace, int ttl) {
+        this.session = session;
+        this.dataAccess = dataAccess;
+        this.ttl = Days.days(ttl).toStandardDuration().getMillis();
+
+        findTables = session.getSession().prepare(
```

Comment:
Other option could be to use the PR #927 findExpiredTables.

Reply:
Doesn't findExpiredTables just return timestamps?

Reply:
Sure, but for those we have ``dataAccess.dropTempTable(startTimeSlice)``

Reply:
mm, actually expired tables is actually just pushing tables that should have been compressed, not dropped completely..